lesson25/index.js

```diff
@@ -0,0 +1,136 @@
+const observerOptions = {
+    root: modalInner,
+    threshold: 1.0
+};
+const checkbox = document.getElementById("js-checkbox");
+
+const setCheckedAttributeToCheckbox = ([entry]) => {
+    if (entry.isIntersecting) {
+        checkbox.checked = true;
+        checkbox.disabled = false;
+    }
+};
+
+const observer = new IntersectionObserver(setCheckedAttributeToCheckbox, observerOptions);
+observer.observe(observerTarget);
+
+
+const nameOfInput = document.querySelector(".js-form-name");
+const emailOfInput = document.querySelector(".js-form-email");
+const passwordOfInput = document.querySelector(".js-form-password");
+const formElements = [nameOfInput, emailOfInput, passwordOfInput];
+
+const validationTerms = {
+    name: {
+        upperLimitOfText: 15
+    },
+    password: {
+        pattern: /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z\d]{8,}$/
+    }
+}
+
+const validationOptions = {
+    name: {
+        isValid: () => {
+            return nameOfInput.value.length < validationTerms.name.upperLimitOfText;
+        },
+        errorMessage: "ユーザー名は15文字以内で入力してください",
+    },
+    email: {
+        isValid: () => {
+            return emailOfInput.validity.valid;
+        },
+        errorMessage: "メールアドレスの形式になっていません",
+    },
+    password: {
+        isValid: () => {
+            return validationTerms.password.pattern.test(passwordOfInput.value);
+        },
+        errorMessage: "8文字以上の大小の英数字を交ぜたものにしてください",
+    }
+};
+
+const addInvalidClass = target => target.classList.add("invalid");
+const removeInvalidClass = target => target.classList.remove("invalid");
+const showErrorMessage = target => target.nextElementSibling.textContent = validationOptions[target.id].errorMessage;
+const isNotEmptyOfInput = target => target.value.trim() !== "";
+const isValidFormInput = target => validationOptions[target.id].isValid();
+
+const isValidInputAndCheckbox = () => {
+    const invalidItem = document.getElementsByClassName("invalid");
+    return invalidItem.length === 0 && checkbox.checked;
+};
+
+const checkFormValidityToEnableSubmitButton = () => submitButton.disabled = isValidInputAndCheckbox() ? false : true;
+
+formElements.forEach(element => {
+    element.classList.add("invalid");
+    
+    element.addEventListener("blur", (e) => {
+        const target = e.target;
+        submitButton.disabled = true;
+
+        if (!isNotEmptyOfInput(target)) {
+            addInvalidClass(target);
+            target.nextElementSibling.textContent = "入力してください";
+            return;
+        }
+
+        if (!isValidFormInput(target)) {
+            showErrorMessage(target);
+            addInvalidClass(target);
+            return;
+        }
+
+        target.nextElementSibling.textContent = "";
+        removeInvalidClass(target);
+        checkFormValidityToEnableSubmitButton();
+    });
+});
+
+checkbox.addEventListener("input", () => {
+    if (!checkbox.checked) {
+        submitButton.disabled = true;
+    }
```

Comment:
IMO
I would write it this way because it always has the opposite value of `checkbox.checked`. But I'll leave it up to you to decide if you want to modify it or not.
```suggestion
submitButton.disabled = !checkbox.checked;
```

Reply:
wow!
Thank you for grate review!!
Very informative.
I fixed it!
[dbbb616](https://github.com/haru-programming/JavaScript-lessons/pull/29/commits/dbbb616c9b1a09efce515580bce62760c60464cc)